File: Tri_insertion/tri_insertion.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int* read_file(char* filename, int* taille){
    FILE* fichier = NULL;
    fichier = fopen(filename, "r");
    int* tab = malloc(*taille * sizeof(int));
    *taille = 0;
    while(fscanf(fichier, "%d",&tab[*taille]) == 1){
        (*taille) ++;
    }
    fclose(fichier);
    return tab;
}

int* tri_tab(int* tab, int taille){
    for(int i=1; i<taille;i++)
    {
        int key = tab[i];
        int j = i-1;
        while(j>=0 && tab[j]>key)
        {
            tab[j+1] = tab[j];
            j=j-1;
        }
        tab[j+1] = key;
    }
    return tab;
}

int* insertion(char* filename, int taille){
    int* tab = read_file(filename,&taille);
    tab = tri_tab(tab,taille);
    for(int i=0; i<taille; i++)
    {
        printf("%d\n", tab[i]);
    }
    return tab;
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Usage: %s <size>\n", argv[0]);
        return 1;
    }

    int size = atoi(argv[1]);
    int* tab = insertion("./tab.txt",size);
    free(tab);

    return 0;
}

File: Tri_insertion/tri_insertion.py
import argparse
import sys

def read_file(filename):
    valeurs = []
    try:
        with open(filename, 'r') as fichier:
            contenu = fichier.read().strip()  
            valeurs = contenu.split()  
            valeurs = [float(val) for val in valeurs]  
    except FileNotFoundError:
        print("Le fichier spécifié n'a pas été trouvé.")
    except Exception as e:
        print("Une erreur est survenue :", e)
    return valeurs

def insertion(filename, size):
    tab = read_file(filename)
    print(tab)
    for i in range(1, size):
        cle = tab[i]
        j = i-1
        while(j>=0 and tab[j] > cle):
            tab[j+1] = tab[j]
            j = j-1
        tab[j+1] = cle

    print(tab)

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Effectue un tri par insertion sur un fichier spécifié.")
    parser.add_argument("filename", type=str, help="Chemin vers le fichier à trier.")
    parser.add_argument("size", type=int, help="Taille du tableau à trier.")
    args = parser.parse_args()

    # Définissez la limite de profondeur de récursion
    sys.setrecursionlimit(args.size)  # Vous pouvez ajuster la limite selon vos besoins

    insertion(args.filename, args.size)
